lib/kafka/async_producer.rb

```diff
@@ -146,16 +151,21 @@ def deliver_messages
     #
     # @see Kafka::Producer#shutdown
     # @return [nil]
-    def shutdown
+    def shutdown(timeout = nil)
       ensure_threads_running!
 
+      @queue.close
```

Comment:
[q] What's the difference between this and `@queue << [:shutdown, nil]`?

Reply:
`@queue.close` will close the queue and prevent anyone else from adding stuff to the queue

`@queue << [:shutdown, nil]` adds a message to the queue, and the consumer will do the shutdown stuff.

This PR changes how the shutdown signal is sent to the consumer, so the changes are that related.